shopfloor/services/zone_picking.py

```diff
@@ -556,15 +556,17 @@ def _scan_source_lot(self, barcode, confirmation=False):
         message = None
         response = None
         search = self._actions_for("search")
-        lot = search.lot_from_scan(barcode)
-        if not lot:
+        # Could get several lots from different products, check each of them
+        lots = search.lot_from_scan(barcode, limit=None)
```

Comment:
why do you specify the limit, as it defaults as None already?

Reply:
Or you could pass 1 and remove the loop to restore the previous behavior.
I have no idea if it could break smth…

Reply:
That's the goal: we have no product to filter lots here, so we could find several lots of different products.
From there we iterate on each lot to search for existing move line linked to this lot:

- if there is one, we return the next screen with `_response_for_set_line_destination()` (success)
- if not, we check the next lot
- once all lots have been checked and no move line has been found, we return `_list_move_lines()` with an error message "Lot not found".

Like before in fact, but before we were expecting one lot corresponding to our barcode... which is not true.

Reply:
And the default value for `limit` here is `1` on purpose:
```def lot_from_scan(self, barcode, products=None, limit=1):```